libvast/src/system/node_control.cpp

```diff
@@ -8,22 +8,47 @@
 
 #include "vast/system/node_control.hpp"
 
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/time.hpp"
 #include "vast/detail/overload.hpp"
 
 #include <caf/scoped_actor.hpp>
+#include <caf/settings.hpp>
 #include <caf/typed_event_based_actor.hpp>
 #include <caf/variant.hpp>
 
 namespace vast::system {
 
+caf::duration node_connection_timeout(const caf::settings& options) {
+  auto timeout = caf::duration{defaults::system::node_connection_timeout};
+  if (auto connection_timeout_arg
+      = caf::get_if<std::string>(&options, "vast.connection-timeout")) {
+    if (auto batch_timeout = to<duration>(*connection_timeout_arg))
+      timeout = caf::duration{*batch_timeout};
+    else
+      VAST_WARN("client cannot set vast.connection-timeout to {} as it "
+                "is not a valid duration",
+                *connection_timeout_arg);
+  }
```

Comment:
Should we create a story for making a helper function for the "get as string and parse afterwards" pattern? (we could also fix it right now, but this sounds like a good small starter story that depends on nothing else)

Reply:
My main worry is about the warning, but I suppose we could have that function return an `expected<duration>` for that purpose. I'm not sure it's worth it, though, considering this is so far only in two places in the code base.

Reply:
Done and implemented.

Reply:
So far it's only used in two places, but I think this is actually the intended behavior in 99% of places where currently `caf::get_if()` is used.